framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+                    // @Pure
+                    // @Tainted String myMethod();
+                    //
+                    // both annotations should be inserted at the same position (the start of
+                    // "String"), but @Pure should always appear first.
+                    if (cmp == 0 && (insertion1.ownLine != insertion2.ownLine)) {
+                        if (insertion1.ownLine) {
+                            cmp = -1;
+                        } else {
+                            cmp = 1;
+                        }
+                    }
+
+                    return -cmp;
+                });
+        StringBuilder result = new StringBuilder(fileContents);
+        for (Insertion insertion : insertions) {
+            result.insert(insertion.position, insertion.contents);
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Inserts all annotations from the ajava file at {@code annotationFilePath} to {@code
+     * javaFilePath}.
+     *
+     * @param annotationFilePath path to an ajava file for {@code javaFilePath}
+     * @param javaFilePath path to a Java file to insert annotation into
+     */
+    public void insertAnnotations(String annotationFilePath, String javaFilePath) {
+        try {
+            Path path = Paths.get(javaFilePath);
+            String fileContents = new String(Files.readAllBytes(path));
+            FileInputStream annotationInputStream = new FileInputStream(annotationFilePath);
+            String result = insertAnnotations(annotationInputStream, fileContents);
+            annotationInputStream.close();
+            Files.write(path, result.getBytes());
+        } catch (IOException e) {
+            System.err.println(
+                    "Failed to insertion annotations from file "
+                            + annotationFilePath
+                            + " to file "
+                            + javaFilePath);
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Inserts annotations from ajava files into Java files in place.
+     *
+     * <p>The first argument is a file or directory containing ajava files. It may be a single ajava
+     * file or a directory containing ajava files.
+     *
+     * <p>The second argument is a Java file or a directory containing Java files to insert
+     * annotations into.
+     *
+     * <p>For each file in the second argument, checks if an ajava file from the first argument
+     * matches it. For each such file, inserts all its annotations into the Java file.
+     *
+     * @param args command line arguments, the first element should be a path to ajava files and the
+     *     second should be the directory containing Java files to insert into
+     */
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.out.println(
+                    "Usage: java InsertAjavaAnnotations <ajava-directory> <java-file-directory");
+            System.exit(1);
+        }
+
```

Comment:
I suggest assigning `args[0]` and `args[1]` into local variables here; their names will be clearer in the following code than the array accesses are.